docs/Consensus/ibft/ibft.md

```diff
@@ -0,0 +1,184 @@
+# IBFT Consensus Overview
+
+## Introduction
+
+Istanbul Byzantine Fault Tolerant (IBFT) consensus is inspired by Castro-Liskov 99 [paper](http://pmg.csail.mit.edu/papers/osdi99.pdf). IBFT inherits from the original PBFT by using a 3-phase consensus, `PRE-PREPARE`, `PREPARE` and `COMMIT`. The system can tolerate at most `F` faulty nodes in a `N` validator network, where `N = 3F + 1`.  
+
+## Implementation
+
+### Terminology
+
+- `Validator`: Block validation participant.
+- `Proposer`: A block validation participant that is chosen to propose block in a consensus round.
+- `Round`: Consensus round. A round starts with the proposer creating a block proposal and ends with a block commitment or round change.
+- `Proposal`: New block generation proposal which is undergoing consensus processing.
+- `Sequence`: Sequence number of a proposal. A sequence number should be greater than all previous sequence numbers. Currently each proposed block height is its associated sequence number.
+- `Backlog`: The storage to keep future consensus messages.
+- `Round state`: Consensus messages of a specific sequence and round, including pre-prepare message, prepare message, and commit message.
+- `Consensus proof`: The commitment signatures of a block that can prove the block has gone through the consensus process.
+- `Snapshot`: The validator voting state from last epoch.
+
```

Comment:
please add, epoch definition: 
The number of blocks (default 30000 blocks) after which to checkpoint and reset the pending votes.

Reply:
EPOCH definition was already present updated documentation to add default blocks